src/Microsoft.Extensions.WebEncoders/Testing/HtmlTestEncoder.cs

```diff
@@ -30,6 +51,21 @@ public override void Encode(TextWriter output, char[] value, int startIndex, int
 
         public override void Encode(TextWriter output, string value, int startIndex, int characterCount)
         {
+            if (output == null)
+            {
+                throw new ArgumentNullException(nameof(output));
+            }
+
+            if (value == null)
+            {
+                throw new ArgumentNullException(nameof(value));
+            }
+
+            if (value.Length == 0)
```

Comment:
That raises another issue: We're not currently validating the `startIndex` and `characterCount` values as the product code would. And this check could short-circuit those validations. One sec while I fix it...